parlai/core/metrics.py

```diff
@@ -278,6 +416,43 @@ def value(self) -> float:
         return sum_ / n
 
 
+class WeightedF1AverageMetric(Metric):
+    """
+    Class that represents the weighted f1 from ClassificationF1Metric
+    """
+
+    __slots__ = '_values'
+
+    def __init__(self, metrics: Dict[str, ClassificationF1Metric]) -> None:
+        self._values = metrics
+
+    def __add__(
+        self, other: Optional['WeightedF1AverageMetric']
+    ) -> 'WeightedF1AverageMetric':
+        if other is None:
+            return self
+        output = dict(**self._values)
+        for k, v in other._values.items():
+            output[k] = output.get(k, None) + v
+        return WeightedF1AverageMetric(output)
```

Comment:
Probably worth taking a look at these mypy warnings

Reply:
yeah i agree, there are a lot of warnings here that are being added, and I don't think that's necessary.